int16 num
int16 time_on